app/Http/Controllers/Admin/ColumnMonsterCrudController.php

```diff
@@ -0,0 +1,910 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use Backpack\CRUD\app\Library\CrudPanel\CrudPanelFacade as CRUD;
+
+/**
+ * Class ColumnMonsterCrudController.
+ *
+ * @property-read \Backpack\CRUD\app\Library\CrudPanel\CrudPanel $crud
+ */
+class ColumnMonsterCrudController extends MonsterCrudController
+{
+    /**
+     * Configure the CrudPanel object. Apply settings to all operations.
+     *
+     * @return void
+     */
+    public function setup()
+    {
+        CRUD::setModel(\App\Models\Monster::class);
+        CRUD::setRoute(config('backpack.base.route_prefix').'/column-monster');
+        CRUD::setEntityNameStrings('column monster', 'column monsters');
+        $this->crud->set('show.setFromDb', false);
+    }
+
+    /**
+     * Define what happens when the List operation is loaded.
+     *
+     * @see  https://backpackforlaravel.com/docs/crud-operation-list-entries
+     *
+     * @return void
+     */
+    public function setupListOperation()
+    {
+        $this->crud->addColumns(static::getColumnsArrayForSimpleTab());
+        $this->crud->addColumns(static::getColumnsArrayForTimeAndSpaceTab());
+        $this->crud->addColumns(static::getColumnsArrayForSelectsTab());
+        $this->crud->addColumns(static::getColumnsArrayForRelationshipsTab());
+        $this->crud->addColumns(static::getColumnsArrayForUploadsTab());
+        $this->crud->addColumns(static::getColumnsArrayForWysiwygEditorsTab());
+        $this->crud->addColumns(static::getColumnsArrayForMiscellaneousTab());
```

Comment:
Hey @munjaldevelopment - I see you've added methods for all the columns, similar to what we have in `MonsterCrudController` for fields. That should be unnecessary - we already have those methods available here, since `ColumnMonsterCrudController extends MonsterCrudController`.

In fact, no only is it unnecessary, it's a shame - by calling those methods directly we'd be testing MORE things. Specifically, we'd be testing that the columns syntaxes match the fields syntaxes. What do I mean by that? If you do `$this->crud->addField(['bla' => 'bla']);` you expect to be able to do EXACTLY `$this->crud->addColumn(['bla' => 'bla']);` and get the corresponding column.

So rather than doing the "hard" thing here and hand-coding all the columns... I think it's much better to do `$this->crud->addColumns($this->getFieldsArrayForRelationshipsTab());`

Then there's no need for all these methods, the code will be much shorter and we'll be testing more things.

Reply:
Hey, Not sure If I follow you. What I am getting is we just need to call $this->crud->addColumns($this->getFieldsArrayForRelationshipsTab()); (only one).  but my question is to test every type of column we need to call all Column types and we can add $this->crud->disableResponsiveTable(); to show all columns on single line.

Please suggest your input for the same.

Reply:
Hey @munjaldevelopment . I'm not saying we should only test one tab. We should test all tabs, definitely. What I'm saying is instead of creating `getColumnsArrayForSimpleTab()`, to use `getFieldsArrayForSimpleTab()` - for all tabs.

So:
```suggestion
        $this->crud->addColumns(static::getFieldsArrayForSimpleTab());
        $this->crud->addColumns(static::getFieldsArrayForTimeAndSpaceTab());
        $this->crud->addColumns(static::getFieldsArrayForSelectsTab());
        $this->crud->addColumns(static::getFieldsArrayForRelationshipsTab());
        $this->crud->addColumns(static::getFieldsArrayForUploadsTab());
        $this->crud->addColumns(static::getFieldsArrayForWysiwygEditorsTab());
        $this->crud->addColumns(static::getFieldsArrayForMiscellaneousTab());
```

Reply:
Ok. Understood & I tried that. But getting error on 
$this->crud->addColumns(static::getFieldsArrayForTimeAndSpaceTab());
as we have
'name'       => ['start_date', 'end_date'], // a unique name for this field

which column is NOT supported. So may be we can create above definition in ColumnMonsterController & yeah rest we can call from MonsterController easily.

What do you think on it? I am getting attached error if I try to call above function.

![image](https://user-images.githubusercontent.com/14837352/194704263-93de4586-efd3-4e31-ab88-5e4a5c410044.png)

Reply:
Applied changes by creating new function for getFieldsArrayForTimeAndSpaceTab in ColumnMonsterCrudController. Request you to please review & revert.